### PR TITLE
NDRS-{323/324}: client query and balance

### PR DESCRIPTION
**Pull request by Fraser999**
This PR is a joint effort between @dwerner and myself to implement the client `query-state` and `get-balance` subcommands.

It is based off https://github.com/CasperLabs/casper-node/pull/232, so only the last 8 commits are relevant.

### CODE REVIEW

**Review by henrytill — COMMENTED** [review]
First pass looks good.  I'll give it a closer look first thing on Monday, but if you get approval before then, feel free to merge.

**Review by EdHastingsCasperLabs — APPROVED** [review]

**Pull request merged**
